convert2rhel/checks.py

```diff
@@ -443,8 +443,15 @@ def get_rhel_supported_kmods():
     #   platform ID.
     # get output of a command to get all packages which are the source
     # of kmods
-    kmod_pkgs_str, _ = run_subprocess(cmd, print_output=False)
+    kmod_pkgs_str, repoquery_exit_code = run_subprocess(cmd, print_output=False)
 
+    if repoquery_exit_code != 0:
+        logger.critical(
+            "We were unable to query the repositories (%s) to"
+            " determine packages containing kernel modules."
```

Comment:
What about adding a note that the reason might be having not enough free disk space? Without any hint like the users would could get easily frustrated, not knowing how to proceed further.

Reply:
The repoquery output which is included on line 452 should include that:
``` 
# repoquery --setopt=module_platform_id=platform:el8 -f '/lib/modules/*.ko*'
Main config did not have a module_platform_id attr. before setopt
Insufficient space in download directory /var/cache/yum/x86_64/7/updates
    * free  4.0 M
    * needed 21 M
```

I could add a little bit that says disk space is one possible reason in case the repoquery output doesn't include that summary.  However, if he repoquery output doesn't include it, my thought is it would be something different than the disk space issue..

What do you think?

Reply:
Hmmm... I haven't yet.  I did see that the error went to stderr which I thought might have caused problems but `run_subprocess()` directs stderr to stdout.  That should have output the repoquery message before though, so I'll double check whether it's at least working with this PR now.

Reply:
Actually... not just not-enabled... not configured at all. there were no repo files referencing those cache directories.

Reply:
And now I'm getting the opposite result (no repodata gives the out of space error) ... However, this makes more sense.... If I have repodata downloaded that says the file belongs to package foo in repo bar but repo bar is disabled, that could just be filtered out, then I could see repoquery saying there are no packages... it's sort of saying "there are no packages with these files [in any of the repositories I have information on]".  OTOH, if there is no repodata downloaded, then repoquery is saying "I need to download data from at least one repository to do anything at all.  You need to free up some space for me to work".

Inside of convert2rhel's run, we will download the repodata for the source os as part of the early checks.  Then, after we enable the RHEL repos, we will do the kmod packages check.  That means that we will always have some repodata when we perform the kmod package check but it won't be for the repository we need.  Thus w will always fall into the first case ("there are no packages with these files [in the repositories I have downloaded metadata for]").

Reply:
Okay, this has now been reworked to run yum makecache before running hte repoquery.  yum makecache will fail if there is not enough disk space with a helpful error message which we can then show to the user.  So hopefully this will work better than repoquery alone.